js/src/wallet/parseURL.test.ts

```diff
@@ -0,0 +1,92 @@
+import { PublicKey } from '@solana/web3.js';
+import BigNumber from 'bignumber.js';
+import { parseURL } from './parseURL';
+
+describe('parseURL', () => {
+    describe('parsing', () => {
+        describe('when given correct params', () => {
+            it('should parse successfully', () => {
+                const url =
+                    'solana:mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN?amount=0.01&reference=82ZJ7nbGpixjeDCmEhUcmwXYfvurzAgGdtSMuHnUgyny&label=Michael&message=Thanks%20for%20all%20the%20fish&memo=OrderId5678';
+
+                const parsed = parseURL(url);
+
+                expect(parsed.recipient).toEqual(new PublicKey('mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN'));
+                expect(parsed.amount).toEqual(new BigNumber(0.01));
+                expect(parsed.token).toBeUndefined();
+                expect(parsed.references).toEqual([new PublicKey('82ZJ7nbGpixjeDCmEhUcmwXYfvurzAgGdtSMuHnUgyny')]);
+                expect(parsed.label).toEqual('Michael');
+                expect(parsed.message).toEqual('Thanks for all the fish');
+                expect(parsed.memo).toEqual('OrderId5678');
+            });
+
+            it('should parse with spl-token', () => {
+                const url =
+                    'solana:mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN?amount=0.01&spl-token=82ZJ7nbGpixjeDCmEhUcmwXYfvurzAgGdtSMuHnUgyny&label=Michael&message=Thanks%20for%20all%20the%20fish&memo=OrderId5678';
+
+                const parsed = parseURL(url);
+
+                expect(parsed.recipient).toEqual(new PublicKey('mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN'));
+                expect(parsed.amount).toEqual(new BigNumber(0.01));
+                expect(parsed.token).toEqual(new PublicKey('82ZJ7nbGpixjeDCmEhUcmwXYfvurzAgGdtSMuHnUgyny'));
+                expect(parsed.references).toBeUndefined();
+                expect(parsed.label).toEqual('Michael');
+                expect(parsed.message).toEqual('Thanks for all the fish');
+                expect(parsed.memo).toEqual('OrderId5678');
+            });
+        });
+    });
+
+    describe('errors', () => {
+        it('throws an error on invalid length', () => {
+            const url = 'X'.repeat(2059);
+            expect(() => parseURL(url)).toThrow('length invalid');
+        });
+
+        it('throws an error on invalid protocol', () => {
+            const url = 'eth:0xffff';
+            expect(() => parseURL(url)).toThrow('protocol invalid');
+        });
+
+        it('throws an error on invalid recepient', () => {
+            const url = 'solana:0xffff';
+            expect(() => parseURL(url)).toThrow('recipient invalid');
+        });
+
+        it('throws an error on missing amount', () => {
+            const url = 'solana:mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN';
+
+            expect(() => parseURL(url)).toThrow('amount missing');
+        });
```

Comment:
This isn't a problem with this test, but we should probably support payment links that don't specify an amount, and the wallet should allow input. This would allow you to use the protocol for transfers between friends like a CashApp QR code.

Reply:
That is very clever

Reply:
Sounds like this is supported by FTX so we should make it part of the spec / impl.

Reply:
nice!

Potentially a different PR for that

Reply:
Yeah sounds good.